include/benchmark/benchmark_api.h

```diff
@@ -239,6 +242,45 @@ inline BENCHMARK_ALWAYS_INLINE void DoNotOptimize(Tp const& value) {
 // FIXME Add ClobberMemory() for non-gnu compilers
 #endif
 
+
+
+class Counter {
+public:
+
+  typedef enum {
+    kDefaults   = 0x00,
+    // Mark the counter as a rate. It will be presented divided by the duration of the benchmark.
+    kIsRate     = 0x01 << 0,
+    // Mark the counter as a thread-average quantity. It will be presented divided by the number of threads.
+    kAvgThreads = 0x01 << 1
+  } Flags;
+
+public:
+
+  double   value;
+  uint32_t flags;
+
+public:
+
+  BENCHMARK_ALWAYS_INLINE Counter(double v, uint32_t flags_) : value(v), flags(flags_) {}
```

Comment:
This class should have an assignment operator `Counter& operator=(double)` to support things like `c = 42.0;`